dev/PipsPager/PipsPager.cpp

```diff
@@ -0,0 +1,555 @@
+    m_selectedIndexChangedEventSource(*this, *args);
+}
+
+winrt::Size PipsPager::GetDesiredPipSize(const winrt::Style& style) {
+    if (auto const repeater = m_pipsPagerRepeater.get())
+    {
+        if (auto const itemTemplate = repeater.ItemTemplate().try_as<winrt::DataTemplate>())
+        {
+            if (auto const element = itemTemplate.LoadContent().try_as<winrt::FrameworkElement>())
+            {
+                element.Style(style);
+                element.Measure({ std::numeric_limits<float>::infinity(), std::numeric_limits<float>::infinity() });
+                return element.DesiredSize();
+            }
+        }
+    }
+    /* Extract default sizes and return in case the code above fails */
+    auto pipHeight = unbox_value<double>(ResourceAccessor::ResourceLookup(*this, box_value(c_pipsPagerButtonHeightPropertyName)));
+    auto pipWidth = unbox_value<double>(ResourceAccessor::ResourceLookup(*this, box_value(c_pipsPagerButtonWidthPropertyName)));
+    return { static_cast<float>(pipWidth), static_cast<float>(pipHeight) };
+}
+
+void PipsPager::OnKeyDown(const winrt::KeyRoutedEventArgs& args) {
+    winrt::FocusNavigationDirection previousPipDirection;
+    winrt::FocusNavigationDirection nextPipDirection;
+    if (Orientation() == winrt::Orientation::Vertical)
+    {
+        previousPipDirection = winrt::FocusNavigationDirection::Up;
+        nextPipDirection = winrt::FocusNavigationDirection::Down;
+    }
+    else
+    {
+        previousPipDirection = winrt::FocusNavigationDirection::Left;
+        nextPipDirection = winrt::FocusNavigationDirection::Right;
+    }
+
+    if (args.Key() == winrt::VirtualKey::Left || args.Key() == winrt::VirtualKey::Up)
+    {
+        winrt::FocusManager::TryMoveFocus(previousPipDirection);
+        args.Handled(true);
+    }
+    else if (args.Key() == winrt::VirtualKey::Right || args.Key() == winrt::VirtualKey::Down)
+    {
+        winrt::FocusManager::TryMoveFocus(nextPipDirection);
+        args.Handled(true);
+    }
+    // Call for all other presses
+    __super::OnKeyDown(args);
+}
+
+void PipsPager::OnPointerEntered(const winrt::PointerRoutedEventArgs& args) {
+    __super::OnPointerEntered(args);
+    m_isPointerOver = true;
+    UpdateNavigationButtonVisualStates();
+}
+void PipsPager::OnPointerExited(const winrt::PointerRoutedEventArgs& args) {
+    // We can get a spurious Exited and then Entered if the button
+    // that is being clicked on hides itself. In order to avoid switching
+    // visual states in this case, we check if the pointer is over the
+    // control bounds when we get the exited event.
+    if (IsOutOfControlBounds(args.GetCurrentPoint(*this).Position()))
+    {
+        m_isPointerOver = false;
+        UpdateNavigationButtonVisualStates();
+    }
+    else
+    {
+        args.Handled(true);
+    }
+    __super::OnPointerExited(args);
+}
```

Comment:
@codendone could you please weight in on this? 
The comment in the function will explain what's happening :)
I'm not sure when to call the **__super** - in the beginning of the function or after - not sure if we ever want to call **__super** with Handled set to true :) (when the onPointerExited was "wronguflly" fired).
It's due to this quirk:
`  if (!CQuirksMode2::QuirkDoNotProcessPointerExitedEventByPointerEnteredElementStateChange() && !bIsEnabled)
        {
            // Process the PointerExited event if the pointer entered element is disabled.
            IFC(ProcessPointerExitedEventByPointerEnteredElementStateChange());
        }`